### PR TITLE
a-crist

### PR DESCRIPTION
**Pull request by nicollis**
Hi Katie,

The bad news: my wife and I came back from the weekend and found some siding was pulled off our house from a storm.

The good news: I had time to do the challenge this morning while waiting for contractors to come by with quotes. 😊

In this email is a link to the zipped project. I have also created a private repo on GitHub - if someone at BNR wants to be added as a collaborator in order to review the code, let me know and I will add them!

Included in the readme of the project are notes concerning some of my assumptions, decisions and other todos.

Let me know if there's anything else you might need from me at this time.

I look forward to hearing back from you!

Thanks!

### CODE REVIEW

**Comment by nicollis** [comment]
# Some quick notes from me on the upload.

## Console warnings from when running the app

- for the tech interview it would be good to bring up `!=` vs `!==` to get a judgement of their JS knowledge because `!=` does have valid uses, and can be an area to see how well they can articulate that technical knowledge.
Screen Shot 2019-06-03 at 6 20 51 PM

## Unable to get anything to pull up.

- someone else should give download this and see what they can get. I would love to see what the UI is suppose to look like. 
Screen Shot 2019-06-03 at 6 20 57 PM

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import rootReducer from './reducers/rootReducer'
+import { createStore, applyMiddleware, compose } from 'redux';
+import { Provider } from 'react-redux';
+import thunk from 'redux-thunk';
+
+const store = createStore(
+  rootReducer,
+  compose(
+    applyMiddleware(thunk),
+    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__())
```
Comment (LivVitale) [inline src/index.js]:
```suggestion
    ...(window.__REDUX_DEVTOOLS_EXTENSION__ ? [window.__REDUX_DEVTOOLS_EXTENSION__()] : [])
```

@nicollis This is the cause of that issue you're seeing. Not having redux devtools installed causes this to fail and crash. (They just needed to not apply middleware in the event that devtools aren't installed.)

Comment (LivVitale) [reply src/index.js]: Some inconsistent formatting in this...

- Lots of missing semicolons
- Inconsistent spacing (`if(` vs `if (`, `const {someImport} =` vs `const { someImport } =`)
- Closing parens alternate between same indentation and end of line (for longer statements)

Mostly minor style gripes easily fixed by Prettier / lint rules.

File: src/actions/appActions.test.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { setPosts, fetchingPosts, handleError, fetchPosts } from './appActions'
+
+// TODO: write fetch test
+
+// it('fetchPosts', async () => {
```
Comment (LivVitale) [inline src/actions/appActions.test.js]: The only use of async/await in the whole codebase and it's commented out. 😭

File: src/reducers/app.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      }
+    case 'SET_POSTS':
+      return {
+        ...state,
+        loading: false,
+        myPosts: action.posts.filter(post => {
+          return post.userId === 1
+        }),
+        othersPosts: action.posts.filter(post => {
+          return post.userId != 1
```
Comment (LivVitale) [inline src/reducers/app.js]: Would be worth checking if this `!=` was intentional vs. `!==`.

File: src/components/app/App.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  componentDidMount() {
+    this.props.fetchPosts()
+  }
+
+  render() {
+    const { myPosts, othersPosts, loading, errorFlag } = this.props
+    return (
+      <div id="app-container">
+      <header id="app-header">Big Nerd Ranch Postification</header>
+      { loading || errorFlag ?
```
Comment (LivVitale) [inline src/components/app/App.js]: Nested ternary... Not super clean and re-checks `loading`, but it does the job. 😁

File: src/components/IndividualPost/IndividualPost.test.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      title: 'title',
+      body: 'body',
+      id: 1
+    }
+    wrapper = shallow(<IndividualPost post={post}/>)
+  });
+
+  describe('When it renders', () => {
+    it('Then there is a title', () => {
+      expect(wrapper.find('.post-title').text()).toEqual('Title: title')
```
Comment (LivVitale) [inline src/components/IndividualPost/IndividualPost.test.js]: These tests are a bit brittle. They'll break if the casing of "Title" changes, if the name of the CSS class used for targeting changes, etc. There are (IMO) better ways to test this (data attribute, select by element), but that's a longer conversation about testing philosophy and not really a fault of this PR.

File: src/components/IndividualPost/index.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+import { IndividualPost } from './IndividualPost'
```
Comment (LivVitale) [inline src/components/IndividualPost/index.js]: Bit confused about this file. There are a couple of these in the project. Almost seemed like barrels at first glance, but they're collecting named exports and re-exporting a single default export...?

If the intent was for this to be a barrel, it would have been nice to have seen named exports.

**Comment by LivVitale** [comment]
Absolutely did not mean for those to be separate reviews... Sorry. 😅 

tl;dr: 
Inconsistencies in code (syntactically/stylistically-speaking - this thing desperately needs to be run through Prettier), a couple of questionable choices (barrels-but-not-actually-barrels?), and some missing tests. Code does what it's intended to, though! Generally good usage of semantically-appropriate tags (except for not making a list of posts a `ul/li`, IMO). Good usage of ES6 syntax (destructuring is super clean!), although I really wish I'd seen some async/await.

**Comment by CodingItWrong** [comment]
@nicollis interestingly i was able to run the app; i wonder what was going on on your end that prevented it 🤷‍♂ - will check out @LivVitale's input and the code a bit later; we are interviewing this candidate friday

Screen Shot 2019-06-05 at 9 27 20 PM
